napari/utils/io.py

```diff
@@ -305,12 +352,121 @@ def read_csv(filename: str) -> Tuple[np.array, List[str]]:
     column_names : list
         List of column names for table data.
     """
-    with open(filename, newline='') as csvfile:
-        reader = csv.reader(csvfile, delimiter=',')
-        output_data = []
-        for row_index, row in enumerate(reader):
-            if row_index == 0:
-                column_names = [str(i) for i in row]
-            else:
-                output_data.append([float(i) for i in row])
-        return np.array(output_data), column_names
+    column_names, *data = iter_csv(filename)
+    return np.array(data), column_names
+
+
+def guess_layer_type_from_column_names(column_names):
```

Comment:
```suggestion
def guess_layer_type_from_column_names(column_names: List[str]) -> Optional[str]:
```
(If you're so inclined 😄 )